src/reducers/bucketlistReducer.js

```diff
@@ -0,0 +1,104 @@
+import {
+  ADD_BUCKETLIST_SUCCESS,
+  ADD_BUCKETLIST_FAILURE,
+  ADD_BUCKETLIST_REQUEST,
+  GET_BUCKETLISTS_REQUEST,
+  GET_BUCKETLISTS_SUCCESS,
+  GET_BUCKETLISTS_FAILURE,
+  UPDATE_BUCKETLIST_REQUEST,
+  UPDATE_BUCKETLIST_SUCCESS,
+  UPDATE_BUCKETLIST_FAILURE,
+  DELETE_BUCKETLIST_SUCCESS,
+  DELETE_BUCKETLIST_FAILURE,
+  DELETE_BUCKETLIST_REQUEST,
+  RESET_REQUEST_STATE
+} from "../actions/constants";
+
+const initialState = {
+  request: {
+    "status": "",
+    "message": ""
+  }
+}
+
+export default function bucketlists(state : Object = initialState, action) {
+  switch (action.type) {
+    
+    case GET_BUCKETLISTS_REQUEST:
+    case UPDATE_BUCKETLIST_REQUEST:
+    case DELETE_BUCKETLIST_REQUEST:
+    case ADD_BUCKETLIST_REQUEST:
+      return state;
+
+    case RESET_REQUEST_STATE:
+      return Object.assign([], state, {
+        request: {
+          status: "",
+          message: ""
+        }
+      });
+
+    case ADD_BUCKETLIST_SUCCESS:
+      return Object.assign([], state, {
+        request: {
```

Comment:
**Question**
I am slightly confused about why we are adding this object here. Does it mean that if we add multiple bucketlists, we'll have multiple objects that look like this in the part of the state tree operated on by this reducer?

Reply:
I'll restructure my initial state to help make it easier to work with but it should be an object of bucketlists if you append to a list it doesn't work.